src/components/images/EditImage.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from "react";
+import LocalApi from "../../apis/LocalApi";
+import { Form, Button, Container } from "react-bootstrap";
+
+class EditImage extends Component {
+  state = {
+    selectedFile: null,
+    lot: 0,
+    unitNumber: "",
+    productDescription: ""
+  };
+
+  //When loading this page, retrive the data for this particular apartment from backend
+  componentDidMount() {
+    LocalApi.get(`${process.env.REACT_APP_BASEURL}/images/`)
+      .then(response => {
+        for (let i = 0; i < response.data.length; i++) {
+          if (response.data[i]._id === this.props.match.params.id) {
+            this.setState({
+              lot: response.data[i].lot,
+              unitNumber: response.data[i].unitNumber,
+              productDescription: response.data[i].productDescription
+            });
+          }
+        }
+      })
+      .catch(function(error) {
+        console.log(error);
+      });
+  }
+
+  imageEditHandler = event => {
+    event.preventDefault();
+    const { lot, selectedFile, unitNumber, productDescription } = this.state;
+
+    LocalApi.post(`/images/edit/${this.props.match.params.id}`, {
+      lot: lot,
+      selectedFile: selectedFile[0].name,
+      unitNumber: unitNumber,
+      productDescription: productDescription
+    })
+      .then(res => {
+        console.log(res.data);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+    this.props.history.push("/");

```

Comment:
Maybe, lets put " this.props.history.push("/");" to LocalApi.post(..).then ? So we redirect user only when we know that form has been  submitted successfully, e.g.

```
LocalApi.post(`/images/edit/${this.props.match.params.id}`, {
      lot: lot,
      selectedFile: selectedFile[0].name,
      unitNumber: unitNumber,
      productDescription: productDescription
    })
      .then(res => {
        console.log(res.data);
        this.props.history.push("/");
      })
      .catch(err => {
        console.log(err);
      });
    
```

Reply:
Just tried... this won't work lol